wpilibc/src/main/native/include/frc/smartdashboard/MechanismObject2d.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <map>
+#include <memory>
+
+#include <wpi/StringMap.h>
+
+#include <networktables/NetworkTable.h>
+
+namespace frc {
+
+class MechanismObject2d {
+  friend class Mechanism2d;
+ protected:
+  explicit MechanismObject2d(const wpi::Twine& name);
+  /**
+   * Update all entries with new ones from a new table.
+   *
+   * @param table the new table.
+   */
+  virtual void UpdateEntries(std::shared_ptr<NetworkTable> table) = 0;
+ public:
+  /**
+   * Retrieve the object's name.
+   * 
+   * @return the object's name relative to its parent.
+   */
+  const std::string& GetName() const;
+
+  /**
+   * Append a Mechanism object that is based on this one.
+   *
+   * @param name the name of the new object.
+   * @param args constructor arguments of the object type.
+   * @return the constructed and appended object, useful for variable assignments and call chaining.
+   * @throw if an object with the given name already exists.
+   */
+  template<typename T, typename... Args>
+  T* Append(wpi::StringRef name, Args&&... args) {    
+    if (m_objects.count(name)) {
+      throw std::runtime_error(("MechanismObject names must be unique! `" + name
+                               + "` was inserted twice!").str());
+    }
+    // return nullptr;
+    std::unique_ptr<T> ptr = std::make_unique<T>(name, std::forward<Args>(args)...);
+    T* ex = ptr.get();
+    m_objects.try_emplace<std::unique_ptr<MechanismObject2d>>(name, std::move(ptr));
+    if (m_table) {
+      ex->Update(m_table->GetSubTable(name));
+    }
+    return ex;
+  }
+
+ private:
+  std::string m_name;
+  wpi::StringMap<std::unique_ptr<MechanismObject2d>> m_objects;
+  std::shared_ptr<NetworkTable> m_table;
+  mutable wpi::mutex m_mutex;
+  void Update(std::shared_ptr<NetworkTable> table);
+
+};
+  bool operator==(MechanismObject2d& a, MechanismObject2d& b);

```

Comment:
Make this const and static since you aren't using `this` anywhere in the implementation.